sqlmesh/dbt/adapter.py

```diff
@@ -98,14 +98,31 @@ def quote(self, identifier: str) -> str:
     def dispatch(self, name: str, package: t.Optional[str] = None) -> t.Callable:
         """Returns a dialect-specific version of a macro with the given name."""
         target_type = self.jinja_globals["target"]["type"]
-        references_to_try = [
-            MacroReference(package=f"{package}_{target_type}", name=f"{target_type}__{name}"),
-            MacroReference(package=package, name=f"{target_type}__{name}"),
-            MacroReference(package=package, name=f"default__{name}"),
+        macro_suffix = f"__{name}"
+
+        def _macro_relevance(val: t.Tuple[str, str]) -> int:
+            """Lower scores more relevant."""
+            package_name, macro_name = val
+            score = 0
+            if package_name != package:
+                score += 2
+            if macro_name.startswith("default"):
+                score += 3
+            elif not macro_name.startswith(target_type):
+                score += 1
+            return score
+
+        relevant_macros = [
+            (package_name, macro_name)
+            for package_name, macros in self.jinja_macros.packages.items()
```

Comment:
Shouldn't we also check `root_macros` if the `package` is None?

Reply:
According to dbt documentation, use the dbt internal packages if no package specified.